app/src/Commands/Instances/CSPCommand.php

```diff
@@ -0,0 +1,153 @@
+<?php declare(strict_types=1);
+
+/* Copyright (c) 2022 - Daniel Weise <daniel.weise@concepts-and-training.de> - Extended GPL, see LICENSE */
+
+namespace CaT\Doil\Commands\Instances;
+
+use CaT\Doil\Lib\Posix\Posix;
+use CaT\Doil\Lib\Docker\Docker;
+use CaT\Doil\Lib\ConsoleOutput\Writer;
+use CaT\Doil\Lib\FileSystem\Filesystem;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+
+class CSPCommand extends Command
+{
+    protected static $defaultName = "instances:csp";
+    protected static $defaultDescription =
+        "This command set CSP rules for the given instance or all instances"
+    ;
+
+    protected Docker $docker;
+    protected Posix $posix;
+    protected Filesystem $filesystem;
+    protected Writer $writer;
+
+    public function __construct(Docker $docker, Posix $posix, Filesystem $filesystem, Writer $writer)
+    {
+        parent::__construct();
+
+        $this->docker = $docker;
+        $this->posix = $posix;
+        $this->filesystem = $filesystem;
+        $this->writer = $writer;
+    }
+
+    public function configure() : void
+    {
+        $this
+            ->setAliases(["csp"])
+            ->addArgument("instance", InputArgument::OPTIONAL, "Name of the instance to start")
+            ->addOption("rules", "r", InputOption::VALUE_REQUIRED, "CSP rules as string")
+            ->addOption("global", "g", InputOption::VALUE_NONE, "Determines if an instance is global or not")
+            ->addOption("all", "a", InputOption::VALUE_NONE, "If is set start all instances")
```

Comment:
Bedeutet -all in diesem Fall nicht, dass die CSP-Regeln für alle Instanzen gesetzt werden? Dann müsste hier sowas stehen wie "If flag is set, apply CSP rules to ALL instances"

Reply:
Das ist erstmal die bekanntmachung aller Flags. Im execute Bereich weiter unten folgt dann der if Teil.